persistence/sql/identity/persister_identity.go

```diff
@@ -666,56 +666,91 @@ func (p *IdentityPersister) ListIdentities(ctx context.Context, params identity.
 
 	con := p.GetConnection(ctx)
 	nid := p.NetworkID(ctx)
-	query := con.Where("identities.nid = ?", nid).Order("identities.id DESC")
 
-	// Credentials are not expanded through `EagerPreload` but manually after
-	// fetching the identities, hence we filter out the relevant expand options.
-	var expandExceptCredentials sqlxx.Expandables
-	for _, e := range params.Expand {
-		if e != identity.ExpandFieldCredentials {
-			expandExceptCredentials = append(expandExceptCredentials, e)
+	joins := ""
+	wheres := ""
+	args := []any{nid}
+	identifier := params.CredentialsIdentifier
+	identifierOperator := "="
+	if identifier == "" && params.CredentialsIdentifierSimilar != "" {
+		identifier = params.CredentialsIdentifierSimilar
+		identifierOperator = "%"
+		switch con.Dialect.Name() {
+		case "postgres", "cockroach":
+		default:
+			identifier = "%" + identifier + "%"
+			identifierOperator = "LIKE"
 		}
 	}
-	if len(expandExceptCredentials) > 0 {
-		query = query.EagerPreload(expandExceptCredentials.ToEager()...)
-	}
 
-	if match := params.CredentialsIdentifier; len(match) > 0 {
+	if len(identifier) > 0 {
 		// When filtering by credentials identifier, we most likely are looking for a username or email. It is therefore
 		// important to normalize the identifier before querying the database.
-		match = NormalizeIdentifier(identity.CredentialsTypePassword, match)
-		query = query.
-			InnerJoin("identity_credentials ic", "ic.identity_id = identities.id").
-			InnerJoin("identity_credential_types ict", "ict.id = ic.identity_credential_type_id").
-			InnerJoin("identity_credential_identifiers ici", "ici.identity_credential_id = ic.id").
-			Where("(ic.nid = ? AND ici.nid = ? AND ici.identifier = ?)", nid, nid, match).
-			Where("ict.name IN (?)", identity.CredentialsTypeWebAuthn, identity.CredentialsTypePassword).
-			Limit(1)
-	} else {
-		query = query.Paginate(params.Page+1, params.PerPage)
-	}
-
-	if err := sqlcon.HandleError(query.All(&is)); err != nil {
-		return nil, err
+		identifier = NormalizeIdentifier(identity.CredentialsTypePassword, identifier)
+
+		joins = `
+INNER JOIN identity_credentials ic ON ic.identity_id = identities.id
+INNER JOIN identity_credential_types ict ON ict.id = ic.identity_credential_type_id
+INNER JOIN identity_credential_identifiers ici ON ici.identity_credential_id = ic.id`
+		wheres = fmt.Sprintf(`
+AND (ic.nid = ? AND ici.nid = ? AND ici.identifier %s ?)
+AND ict.name IN (?, ?)`, identifierOperator)
+		args = append(args, nid, nid, identifier, identity.CredentialsTypeWebAuthn, identity.CredentialsTypePassword)
+	}
+
+	// Follow up: add page token support here, will be easy.
+	paginator := pop.NewPaginator(params.Page, params.PerPage)
+
+	if err := con.RawQuery(fmt.Sprintf(`SELECT DISTINCT identities.*
```

Comment:
This query is now build here because pop does not allow `DISTINCT` queries. We need this, because in many cases the query would return an identity twice, once for the `webauthn` and once for the `password` credential type.